### DISCUSSION TITLE
Project schedule

### DISCUSSION THREAD

**Discussion by valbers** [fsprojects/FSharp.Data.GraphQL#431] [Ideas]
I'm thinking about how we can bring some update regularity into this project, so I want to share an idea with you and hear your thoughts.
I guess everybody involved would agree that it would be nice if there were a mechanism that would make sure that the "clock ticks" in this project and that in a way that is not overly dependant on the individual contributors.
To be more specific:
* how long is it acceptable for a pull request to remain open without being either accepted or rejected?
* how long is it acceptable for a pull requester to not react to comments/suggestions to his pull request before it is rejected?
* how long is it acceptable for an issue to be unanswered and open?
* how long until a next patch/minor/major version is released?

Basically, the idea is to define a state machine (which is not necessarily automated) in which every state transition is accompanied by a timeout default transition. Of course, as soon as we introduce something like this we'll probably also want to introduce a set of default or usual acceptance criteria for these states. For example, we might require a certain test coverage threshold for code changes.

**Comment by njlr** [comment]
More generally, I would like to see a more "little and often" approach to releases. I hope that @xperiandri 's huge efforts recently will lay the groundwork for this!

**Reply by xperiandri** [reply]
Yep, we are ready to that

**Reply by xperiandri** [reply]
Around 2 next weeks, I suppose we will have 2.0 and then work incrementally

**Reply by xperiandri** [reply]
The only thing that is now in disbalance is that we have GitHub workflows and `Build.fsx`.
And `Build.fsx` had some packaging logic.
So I am in doubts whether should we use `Build.fsx` for all or not

**Reply by valbers** [reply]
The nice thing about Build.fsx is that you can run it locally. I use it when I make my changes to be sure that I didn't break anything.

**Reply by xperiandri** [reply]
But I have an issue, that FAKE does not work with .NET 7 SDK

**Reply by njlr** [reply]
Could FAKE be replaced by a simple bash script that calls `dotnet`?

Or even a `Makefile`?

The project build is not very complex.

**Reply by valbers** [reply]
I didn't take the time yet, but it looks like there are some workaround suggested here: https://github.com/fsprojects/FAKE/issues/2719
